File: API/user-service/src/main/java/com/friendbook/userservice/DTO/UserProfile.java
package com.friendbook.userservice.DTO;

import java.util.List;

import com.friendbook.userservice.model.User;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserProfile {
    private String name;
    private String login;
    private String image;
    private String email;
    private List<Object> reviews;
    private List<Object> ratedBooks;
    private List<Object> savingBooks;
    private int countFriends;

    public UserProfile(User user) {
        this.name = user.getName();
        this.login = user.getLogin();
        this.email = user.getEmail();
    }
}

File: client/app/src/main/java/com/src/book/domain/model/user/UserProfile.kt
package com.src.book.domain.model.user

import com.src.book.domain.model.BookList
import com.src.book.domain.model.userReview.UserReview

data class UserProfile(
    val name: String,
    val login: String,
    val email: String,
    val image:String,
    val reviews: List<UserReview>?,
    val ratedBooks: List<BookList>?,
    val savingBooks: List<BookList>?,
    val countFriends:Long
)